src/domain/release-archive.ts

```diff
@@ -55,34 +52,37 @@ export class ReleaseArchive {
     url: string,
     stripPrefix: string
   ): Promise<ReleaseArchive> {
-    const filename = url.substring(url.lastIndexOf('/') + 1);
-    const downloadedPath = path.join(
-      fs.mkdtempSync(path.join(os.tmpdir(), 'archive-')),
-      filename
-    );
-    await download(url, downloadedPath);
+    const artifact = new Artifact(url);
+
+    try {
+      await artifact.download();
+    } catch (e) {
+      if (e instanceof ArtifactDownloadError) {
+        throw new ArchiveDownloadError(e.url, e.statusCode);
```

Comment:
nit: curious why it's not just re-throw e?

Reply:
This error type gives a bit more meaning (it's not just some artifact error, it's specifically the release archive). This allows for some nicer error messaging and tips for the user to fix it.